### DISCUSSION TITLE
Task ser/des not working for multiple return values

### DISCUSSION THREAD

**Discussion by mpgreg** [apache/airflow#31680] [Q&A]
### Apache Airflow version

2.6.1

### What happened

Returning multiple values from a taskflow task results in 

`File "/usr/local/lib/python3.10/site-packages/airflow/models/xcom_arg.py", line 263, in __iter__
    raise TypeError("'XComArg' object is not iterable")
TypeError: 'XComArg' object is not iterable`

### What you think should happen instead

Returning multiple values should unpack as a tuple

### How to reproduce

```
@dag(schedule=None, start_date=datetime(2023, 1, 1), catchup=False)
def testdag():
    
    @task()
    def test():  
        return 'x', 'y'
    _x, _y = test()

testdag()
```

### Operating System

5.15.49-linuxkit

### Versions of Apache Airflow Providers

apache-airflow-providers-amazon==8.0.0
apache-airflow-providers-celery==3.1.0
apache-airflow-providers-cncf-kubernetes==6.1.0
apache-airflow-providers-common-sql==1.4.0
apache-airflow-providers-datadog==3.2.0
apache-airflow-providers-elasticsearch==4.4.0
apache-airflow-providers-ftp==3.3.1
apache-airflow-providers-google==10.0.0
apache-airflow-providers-http==4.3.0
apache-airflow-providers-imap==3.1.1
apache-airflow-providers-microsoft-azure==6.1.0
apache-airflow-providers-postgres==5.4.0
apache-airflow-providers-redis==3.1.0
apache-airflow-providers-snowflake==4.1.0
apache-airflow-providers-sqlite==3.3.2

### Deployment

Astronomer

### Deployment details

Astro runtime 8.3.0 Tested only in local dev mode

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by boring-cyborg** [comment]
Thanks for opening your first issue here! Be sure to follow the issue template! If you are willing to raise PR to address this issue please do so, no need to wait for approval.

**Comment by hussein-awala** [comment]
This is not a bug. It is not possible to unpack an Xcom in parsing time because it does not exist at that moment. Xcom is created at runtime after the first task is executed:
```python
from datetime import datetime

from airflow.decorators import dag, task

@dag(schedule=None, start_date=datetime(2023, 1, 1), catchup=False)
def testdag():

    @task()
    def test():
        return 'x', 'y'

    @task()
    def process_x(values):
        x = values[0]
        print(x)

    @task()
    def process_y(values):
        y = values[1]
        print(y)

    returned_values = test()
    process_x(returned_values)
    process_y(returned_values)

testdag()
```

**Reply by ilharco** [reply]
@hussein-awala How did you solve your "issue", may I ask? I'm facing similar issue, but with a task (translate to regular Python Operator) that gets context and returns a dict (I get context["params] to get the passed params).
I'm using TaskFLowAPI, but for this in particular, I'm unable to succeed.

**Reply by ilharco** [reply]
I was able to get the context params via a task

```
@task(task_id="trying_params")
def get_data(params=None):
    print(f"params ({type(params)}) = {params}")
    return params
```
    
and I pass it down to downstream task, whihc an external task: works fine. The problem is when I try the same but calling a task_group that takes the returned dict , same I did with task1 >>task2 , does not work even iterating through same dict, giving this error: 
 `    for (source_obj, target_obj) in zip(dag_params['SOURCE_OBJ'], dag_params['TARGET_OBJ']): TypeError: 'XComArg' object is not iterable`
